File: 3737-count-subarrays-with-majority-element-i/3737-count-subarrays-with-majority-element-i.java
class Solution {
    public int countMajoritySubarrays(int[] nums, int target) {
        int n = nums.length;

        int[] pref = new int[n + 1];
        for (int i = 0; i < n; i++) {
            pref[i + 1] = pref[i] + (nums[i] == target ? 1 : -1);
        }

        int ans = 0;
        for (int i = 0; i <= n; i++) {
            for (int j = i + 1; j <= n; j++) {
                if (pref[j] > pref[i]) {
                    ans++;
                }
            }
        }

        return ans;
    }
}

File: 3737-count-subarrays-with-majority-element-i/3737-count-subarrays-with-majority-element-i.py
class Solution:
    def countMajoritySubarrays(self, nums: List[int], target: int) -> int:
        n = len(nums)

        pref = [0] * (n + 1)
        for i in range(n):
            pref[i + 1] = pref[i] + (1 if nums[i] == target else -1)

        ans = 0
        for i in range(n + 1):
            for j in range(i + 1, n + 1):
                if pref[j] > pref[i]:
                    ans += 1

        return ans